src/main.rs

```diff
@@ -203,18 +233,18 @@ async fn main() -> anyhow::Result<()> {
             WorkerType::Reader => {
                 n_readers_finished += 1;
 
-                if let Ok((query_timings, total_time)) = res {
-                    let qps = query_timings.len() as f64 / total_time.as_secs_f64();
+                if let Ok(query_timings) = res {
+                    let reader_qps = query_timings.values().sum::<f64>() / query_timings.len() as f64;
```

Comment:
what does query_timings.values() store?

Reply:
just the values of the (query id -> avgqps) map; So this is a list of the avgqps for all queries

Reply:
Do you mean the QPS of all queries?

Reply:
for each query its the average of all QPS measurements (as each query might be run multiple times)

Reply:
regarding the hole (time -> inv -> avg) vs (time -> avg -> inv) discussion, Charlotte recommended this: https://math.stackexchange.com/questions/3107661/why-is-the-inverse-of-an-average-of-numbers-not-the-same-as-the-average-of-the-i

I don't have the time right now to completely wrap my head around it, though. So, no real recommendation from my side. 

For how IGUANA does it, I summon @nck-mlcnv :)

Reply:
I have now implemented the time -> avg -> inv because supposedly iguana does it that way. Could you double check, this is what we wanted/its correct, @nkaralis?

Reply:
looks good